src/bit.c

```diff
@@ -159,16 +166,18 @@ int bit_load_firmware(struct bit* bit, struct cha* cha, struct chb* chb) {
 	while (length) {
 		size_t i = 0;
 
-		for (i = 0; i < sizeof(buf) && length; i++, length--, data++) {
-			buf[i] = bitreverse8(*data);
+		bidx = (bidx + 1) % 2;
+
+		for (i = 0; i < sizeof(buf[bidx]) && length; i++, length--, data++) {
+			buf[bidx][i] = bitreverse8(*data);
 		}
 
 		if (tc && ftdi_transfer_data_done(tc) < 0) {
 			bit->error_str = ftdi_get_error_string(&cha->ftdi);
 			return -1;
 		}
 
-		if (!(tc = ftdi_write_data_submit(&cha->ftdi, buf, i))) {
+		if (!(tc = ftdi_write_data_submit(&cha->ftdi, buf[bidx], i))) {
```

Comment:
Alternative solution is to simply change the call to ftdi_write_data_submit() with ftdi_write_data().